src/pybind/ceph_volume_client.py

```diff
@@ -439,19 +441,33 @@ def create_volume(self, volume_path, size=None, data_isolated=False):
         path = self._get_path(volume_path)
 
         self._mkdir_p(path)
-
         if size is not None:
             self.fs.setxattr(path, 'ceph.quota.max_bytes', size.__str__(), 0)
 
+        done = False;
         if data_isolated:
-            pool_name = "{0}{1}".format(self.POOL_PREFIX, volume_path.volume_id)
-            pool_id = self._create_volume_pool(pool_name)
-            mds_map = self._rados_command("mds dump", {})
-            if pool_id not in mds_map['data_pools']:
-                self._rados_command("mds add_data_pool", {
-                    'pool': pool_name
-                })
-            self.fs.setxattr(path, 'ceph.dir.layout.pool', pool_name, 0)
+            isolate_by_namespace = bool(self.rados.conf_get("cephFSClient_data_isolate_by_namespace"))
+            if isolate_by_namespace:
```

Comment:
I think it would be better to leave it inside "data_isolated" block, as user may want to create a public share(volume) with no namespace specified(or a ns called "public") , and in the authorization by default grant user access to the public namespace. Is that reasonable?